pkg/dns/aws/dns.go

```diff
@@ -190,31 +190,25 @@ func (m *Manager) getLBHostedZone(name string) (string, error) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	if id, exists := m.lbZones[name]; exists {
-		return id, nil
+	if zoneID, exists := m.lbZones[name]; exists {
+		return zoneID, nil
 	}
 
-	var id string
-	fn := func(resp *elb.DescribeLoadBalancersOutput, lastPage bool) (shouldContinue bool) {
-		for _, lb := range resp.LoadBalancerDescriptions {
-			log.V(0).Info("found load balancer", "name", aws.StringValue(lb.LoadBalancerName), "dns name", aws.StringValue(lb.DNSName), "hosted zone ID", aws.StringValue(lb.CanonicalHostedZoneNameID))
-			if aws.StringValue(lb.CanonicalHostedZoneName) == name {
-				id = aws.StringValue(lb.CanonicalHostedZoneNameID)
-				return false
-			}
-		}
-		return true
-	}
-	err := m.elb.DescribeLoadBalancersPages(&elb.DescribeLoadBalancersInput{}, fn)
+	// An elb id is the first 32 characters of the elb's dns name.
+	lbID := name[0:32]
```

Comment:
Interesting... how safe is it to make this assumption? Is it documented somewhere?

Reply:
From [elb docs](https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CreateLoadBalancer.html):

> LoadBalancerName
This name must be unique within your set of load balancers for the region, must have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, and cannot begin or end with a hyphen.

From cloud provider [code](https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L86-L89):
```
	//AWS requires that the name of a load balancer is shorter than 32 bytes.
	if len(ret) > 32 {
		ret = ret[:32]
	}
```

Should we instead call `DefaultLoadBalancerName()` in the cloudprovider pkg to get the lb name? If so, I believe it saves us an AWS API call.

Reply:
Hm, so the AWS cloud provider [looks up the load balancer](https://github.com/kubernetes/cloud-provider-aws/blob/8f6e1e3d2cfa20a0deac088b9c4022d433500369/pkg/cloudprovider/providers/aws/aws.go#L3651) using [a name](https://github.com/kubernetes/cloud-provider-aws/blob/8f6e1e3d2cfa20a0deac088b9c4022d433500369/pkg/cloudprovider/providers/aws/aws.go#L3667) that is ultimately [derived from the Service name](https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L86-L89).  If we can assume that the cloud provider code is correct, we can simplify our code some.  I don't see any implication that the load balancer name must match the hosted zone id though, so I think we still need a `DescribeLoadBalancers` API call to get the hosted zone id though.

Reply:
> Hm, so the AWS cloud provider looks up the load balancer using a name that is ultimately derived from the Service name.

That's what I see in the code and when I look at the service id and status.loadbalancer.ingress.hostname.

> If we can assume that the cloud provider code is correct, we can simplify our code some.

OK, let me know if you have any suggestions beyond this PR.